Add basic policy for BinderFS

Conan-Kudo opened the pull request:
Binder is available in Fedora kernels since 5.15, but needs a
basic policy for BinderFS.

This policy lays the groundwork for Waydroid (a containerized
Android runtime environment for regular Linux systems) to function.

Co-authored-by: @aleasto
Co-authored-by: @Conan-Kudo

Conan-Kudo commented:
@zpytela Could you please take a look at this and get it merged? It's needed for proceeding on Waydroid enablement in Fedora.

zpytela commented:
The `_device_t` part should go to devices.te. I'd also appreciate some (short) comment or a resource link to understand a bit more.

Conan-Kudo commented:
Updated as you asked.

Here are a couple of useful resources to understand BinderFS:

* Android BinderFS documentation: https://www.kernel.org/doc/html/latest/admin-guide/binderfs.html
* @brauner's write-up on BinderFS and its history for Linux and how it works: https://brauner.github.io/2019/01/09/android-binderfs.html

zpytela commented:
I meant also the genfscon binder_device_t statements.

zpytela commented:
How about adding
https://www.kernel.org/doc/html/latest/admin-guide/binderfs.html
to the commit message?

Conan-Kudo commented:
How is that?

Conan-Kudo commented:
Done, how is that for you?

zpytela commented:
@Conan-Kudo the genfscon binder_device_t statements should be in devices.te.

Conan-Kudo committed:
Add basic policy for BinderFS

Binder is available in Fedora kernels since 5.15, but needs a
basic policy for BinderFS.

This policy lays the groundwork for Waydroid (a containerized
Android runtime environment for regular Linux systems) to function.

Reference: https://www.kernel.org/doc/html/latest/admin-guide/binderfs.html

Co-authored-by: Alessandro Astone <ales.astone@gmail.com>
Co-authored-by: Neal Gompa <ngompa@fedoraproject.org>

Conan-Kudo committed:
devices: Add a comment about cardmgr_dev_t

It's missing a comment describing it for PCMCIA manager devices

Conan-Kudo commented:
Done, I've also added a comment about `cardmgr_dev_t` in devices.te as a separate commit to make it look less confusing

zpytela commented:
@WOnder93 I'd appreciate your take on these changes.

aleasto commented on policy/modules/kernel/devices.te:
```diff
@@ -46,6 +46,24 @@ dev_node(apm_bios_t)
+type binder_device_t;
+dev_node(binder_device_t)
+# Default standard Linux binderfs control device (/dev/binderfs)
+genfscon binder /binderfs gen_context(system_u:object_r:binder_device_t,s0)
```
This line should not exist. It is pointing to /dev/binderfs/binderfs

Conan-Kudo replied:
gah, fixed.

WOnder93 reviewed (approved):
LGTM

zpytela commented:
Merging, thank you.

Conan-Kudo commented:
Thanks for merging it, @zpytela! Can we get this shipped in F36+?
